src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -143,7 +141,30 @@ def _expand_config_op_types(self, config):
                 expanded_op_types.append(op_type)
         return expanded_op_types
 
-
+class PrunerLayerWrapper(torch.nn.Module):
+    def __init__(self, module, name, type, config, pruner):
+        super().__init__()
+        # origin layer information
+        self.module = module
+        self.name = name
+        self.type = type
+        # config and pruner
+        self.config = config
+        self.pruner = pruner
+        # register buffer for mask
+        self.register_buffer("weight_mask", None)
+        self.register_buffer("bias_mask", None)
+
+    def forward(self, *input):
+        self.pruner.calc_mask(self)
```

Comment:
pruner class and wrapper class are using each other.

Reply:
Yes. Pruner needs to access wrapper module's mask and wrapper module needs to call pruner's cal_mask function.

Reply:
one option is that instead of input `self`, input explicitly input the arguments that `calc_mask` uses.